test/e2e/upgrades/upgrade.go

```diff
@@ -41,6 +41,9 @@ const (
 
 // Test is an interface for upgrade tests.
 type Test interface {
+	// Name should return a test name sans spaces.
+	Name() string
```

Comment:
Nevermind about the framework array, I missed that the name gets passed to the framework constructor.

I was just thinking that using something like `reflect.TypeOf(test)` would reduce the methods each test needed to implement by 1. Not super important though.